crypto/cmp/cmp_msg.c

```diff
@@ -593,23 +593,16 @@ OSSL_CMP_MSG *ossl_cmp_rp_new(OSSL_CMP_CTX *ctx, const OSSL_CMP_PKISI *si,
         goto err;
     rep = msg->body->value.rp;
 
-    if ((si1 = OSSL_CMP_PKISI_dup(si)) == NULL)
+    if ((si1 = OSSL_CMP_PKISI_dup(si)) == NULL
+            || !sk_OSSL_CMP_PKISI_push(rep->status, si1))
         goto err;
```

Comment:
After a successful push shouldn't we now set `si1` to NULL to prevent a double free (when `msg` gets freed) if a subsequent error occurs?

Reply:
Absolutely. Thanks.